src/tests/D3MHub.t.sol

```diff
@@ -881,95 +808,91 @@ contract D3MHubTest is DssTest {
         uint256 prevDai = vat.dai(vow);
 
-        d3mHub.exec(ilk);
+        hub.exec(ilk);
 
-        (, art) = vat.urns(ilk, address(d3mTestPool));
+        (, art) = vat.urns(ilk, address(pool));
         assertEq(art, 50 * WAD);
         uint256 currentDai = vat.dai(vow);
         assertEq(currentDai, prevDai + 10 * RAD); // Interest shows up in vat Dai for the Vow [rad]
         // Make sure pre/post functions get called
-        assertTrue(d3mTestPool.preDebt());
-        assertTrue(d3mTestPool.postDebt());
+        assertTrue(pool.preDebt());
+        assertTrue(pool.postDebt());
     }
 
     function test_exec_fees_not_enough_liquidity() public {
         _windSystem();
         // interest is determined by the difference in gem balance to dai debt
         // by giving extra gems to the Join we simulate interest
-        _giveTokens(TokenLike(address(testGem)), 10 * WAD);
-        testGem.transfer(address(d3mTestPool), 10 * WAD); // Simulates 10 WAD of interest accumulated
+        address(testGem).setBalance(address(this), 10 * WAD);
+        testGem.transfer(address(pool), 10 * WAD); // Simulates 10 WAD of interest accumulated
+        assertEq(pool.assetBalance(), 60 * WAD);
 
-        (uint256 ink, uint256 art) = vat.urns(ilk, address(d3mTestPool));
+        (uint256 ink, uint256 art) = vat.urns(ilk, address(pool));
         assertEq(ink, 50 * WAD);
         assertEq(art, 50 * WAD);
         uint256 prevDai = vat.dai(vow);
 
         // If we do not have enough liquidity then we pull out what we can for the fees
         // This will pull out all but 2 WAD of the liquidity
-        assertEq(dai.balanceOf(address(testGem)), 50 * WAD); // liquidity before simulating other user's withdraw
-        testGem.giveAllowance(address(dai), address(this), type(uint256).max);
-        dai.transferFrom(address(testGem), address(this), 48 * WAD);
-        assertEq(dai.balanceOf(address(testGem)), 2 * WAD); // liquidity after
+        assertEq(dai.balanceOf(address(pool)), 50 * WAD); // liquidity before simulating other user's withdraw
+        pool.decreaseIdleLiquidity(48 * WAD);
+        assertEq(dai.balanceOf(address(pool)), 2 * WAD);  // liquidity after
 
-        d3mHub.exec(ilk);
+        hub.exec(ilk);
 
-        (ink, art) = vat.urns(ilk, address(d3mTestPool));
+        (ink, art) = vat.urns(ilk, address(pool));
         // Collateral and debt increase by 8 WAD as there wasn't enough liquidity to pay the fees accumulated
         assertEq(ink, 58 * WAD);
         assertEq(art, 58 * WAD);
-         // 10 RAY immediately shows up in the surplus
         assertEq(vat.dai(vow), prevDai + 10 * RAD);
         // Make sure pre/post functions get called
-        assertTrue(d3mTestPool.preDebt());
-        assertTrue(d3mTestPool.postDebt());
+        assertTrue(pool.preDebt());
+        assertTrue(pool.postDebt());
     }
 
     function test_exit() public {
         _windSystem();
+        pool.decreaseIdleLiquidity(50 * WAD);
```

Comment:
I see this is being used here to have token availability in the pool as in the Mock we are not minting anymore. It feels a bit hacky though, and I think this comes also related to my previous comment of what increase/decreaseIdleLiquidity. Wondering if there is a bit better way to handle that, maybe more similar to how the mock was before. Did you see any major problem with that model?

Reply:
I think this is a more minimal version of the mock. DAI comes in via `pool.deposit()` and later goes out to somewhere else to be replaced with another token. Could be aDAI, cDAI, but it's also valid for this to just be 1 USDC or something. Maybe the name of the function `decreaseIdleLiquidity()` is confusing?

Reply:
I understand you want to make an abstraction, but then something like `test_exit` looks very weird. As why you need to decrease the liquidity in order to have the exit token. That should be there always regardless how much DAI there is.
I'm proposing this PR which sets the things a bit more similar than before but keeping the Mock pretty simple: https://github.com/makerdao/dss-direct-deposit/pull/108.
If you feel that this breaks the abstraction you want to get, then let's keep discussing.

Reply:
So at its core there is no requirement in `deposit()`/`withdraw()` to convert into another token right away. This is what happens in Aave/Compound, but for a PSM swap the pool can be filled with some mix of the gem and the dai as you can see here: https://github.com/makerdao/dss-direct-deposit/pull/105/files#diff-09fcb804207fcd0316dc1b3f8a584c9097581115543a1f2e561f0e4fc5ab6552R203-R209

The checks in the tests that the amount of gem balance has increased is specific to how Aave/Compound-style works and so that's why I'm trying to keep it as minimal as possible. Basically the pool is a black box where you stick some DAI in and then some random actions will occur in the future to change the mix between `dai` and the `redeemable()` token. This is how I'm modeling the increase/decrease idle dai functions. IE something happened, could be a swap, could be a borrow, could be a withdraw, doesn't matter. The point is the DAI available to immediately withdraw has dropped and some other asset has increased.

Does it clear it up?